src/solc_0.8/polygon/child/land/PolygonLandBaseToken.sol

```diff
@@ -0,0 +1,387 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import "@openzeppelin/contracts-0.8/utils/Address.sol";
+import "../../../common/BaseWithStorage/ERC721BaseToken.sol";
+
+contract PolygonLandBaseToken is ERC721BaseToken {
+    using Address for address;
+
+    uint256 internal constant GRID_SIZE = 408;
+
+    uint256 internal constant LAYER = 0xFF00000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_1x1 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_3x3 = 0x0100000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_6x6 = 0x0200000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_12x12 = 0x0300000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_24x24 = 0x0400000000000000000000000000000000000000000000000000000000000000;
+
+    /**
+     * @notice Return the name of the token contract
+     * @return The name of the token contract
+     */
+    function name() external pure returns (string memory) {
+        return "Sandbox's LANDs";
+    }
+
+    /**
+     * @notice Return the symbol of the token contract
+     * @return The symbol of the token contract
+     */
+    function symbol() external pure returns (string memory) {
+        return "LAND";
+    }
+
+    //check if I can get this from some other place
+    // solium-disable-next-line security/no-assign-params
+    function uint2str(uint256 _i) private pure returns (string memory _uintAsString) {
+        if (_i == 0) {
+            return "0";
+        }
+
+        uint256 j = _i;
+        uint256 len;
+        while (j != 0) {
+            len++;
+            j /= 10;
+        }
+
+        bytes memory bstr = new bytes(len);
+        uint256 k = len - 1;
+        while (_i != 0) {
+            bstr[k--] = bytes1(uint8(48 + (_i % 10)));
+            _i /= 10;
+        }
+
+        return string(bstr);
+    }
```

Comment:
It's seems there was a bug on this function could you see if you could use :
https://stackoverflow.com/a/65707309

Reply:
nice catch